exec.go

```diff
@@ -234,7 +234,6 @@ func getProcess(context *cli.Context, bundle string) (*specs.Process, error) {
 			p.Capabilities.Bounding = append(p.Capabilities.Bounding, c)
 			p.Capabilities.Effective = append(p.Capabilities.Effective, c)
 			p.Capabilities.Permitted = append(p.Capabilities.Permitted, c)
-			p.Capabilities.Ambient = append(p.Capabilities.Ambient, c)
```

Comment:
I think we should add a check here:
```
if p.Capabilities.Inheritable != nil {
  p.Capabilities.Ambient = append(p.Capabilities.Ambient, c)
}
```
Or else it will cause a compatibility issue.

Furthermore, maybe the patch of `[GHSA-f3fp-gc8g-vw66](https://github.com/opencontainers/runc/security/advisories/GHSA-f3fp-gc8g-vw66)` also should be like this:
```
if p.Capabilities.Inheritable != nil {
  p.Capabilities.Inheritable = append(p.Capabilities.Inheritable, c)
  p.Capabilities.Ambient = append(p.Capabilities.Ambient, c)
}
```
But this is not very important, because It depends on our understanding of `runc exec --cap`: `add a capability to the bounding set for the process`. The important one is the compatibility issue metioned above.

Reply:
Thanks for raising this!

Since [runc 1.1.2](https://github.com/opencontainers/runc/releases/tag/v1.1.2) we do not set inheritable capabilities in `runc exec --cap` (and it also means we do not set any ambient capabilities either), and we have yet to receive a bug report.

So perhaps this is now normal?

Reply:
> the documentation also says `runc exec --caps` adds a cap to the bounding set.

How do you understand this sentence? Only add to bounding set?



> I think we should add a check here:
> 
> ```
> if p.Capabilities.Inheritable != nil {
>   p.Capabilities.Ambient = append(p.Capabilities.Ambient, c)
> }
> ```
> 
> Or else it will cause a compatibility issue.

If you can add this check to ensure we can be compatible with previous, I think it will be better.

Reply:
Adding this test here.

Reply:
Added the test case from #4406 and the conditional setting of ambient caps in runc exec --caps.